src/main/java/org/elasticsearch/action/admin/indices/optimize/TransportOptimizeAction.java

```diff
@@ -121,11 +121,11 @@ protected GroupShardsIterator shards(ClusterState clusterState, OptimizeRequest
 
     @Override
     protected ClusterBlockException checkGlobalBlock(ClusterState state, OptimizeRequest request) {
-        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA);
+        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
```

Comment:
When I updated this block I supposed that users want to block refresh/optimize actions on a read only cluster so I set `METADATA_WRITE`. Anyway I don't have a strong opinion on this and `METADATA_READ` is fine to me since it will still be blockable with the `index.blocks.metadata` setting.